File: app/src/main/java/ru/p3tr0vich/fuel/utils/UtilsFileIO.java
package ru.p3tr0vich.fuel.utils;

import android.support.annotation.NonNull;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.List;

public class UtilsFileIO {

    public static void read(@NonNull File file, @NonNull List<String> strings) throws IOException {
        FileInputStream fileInputStream = new FileInputStream(file);

        InputStreamReader inputStreamReader = new InputStreamReader(fileInputStream);

        BufferedReader bufferedReader = new BufferedReader(inputStreamReader);

        String string = bufferedReader.readLine();
        while (string != null) {
            strings.add(string);
            string = bufferedReader.readLine();
        }

        bufferedReader.close();

        inputStreamReader.close();

        fileInputStream.close();
    }

    public static void write(@NonNull File file, @NonNull List<String> strings) throws IOException {
        FileOutputStream fileOutputStream = new FileOutputStream(file);

        OutputStreamWriter outputStreamWriter = new OutputStreamWriter(fileOutputStream);

        BufferedWriter bufferedWriter = new BufferedWriter(outputStreamWriter);

        for (String s : strings) {
            bufferedWriter.write(s);
            bufferedWriter.newLine();
        }

        bufferedWriter.close();

        outputStreamWriter.close();

        fileOutputStream.close();
    }

    public static void makeDir(@NonNull File dir) throws IOException {
        if (dir.exists()) return;

        if (!dir.mkdirs())
            throw new IOException("UtilsFileIO -- makeDir: can not create dir " + dir.toString());
    }

    public static void deleteFile(@NonNull File file) throws IOException {
        if (!file.exists()) return;

        if (!file.delete())
            throw new IOException("UtilsFileIO -- deleteFile: can not delete file " + file.toString());
    }

    public static void createFile(@NonNull File file) throws IOException {
        if (file.exists()) {
            if (!file.isFile())
                throw new FileNotFoundException("UtilsFileIO -- createFile: " + file.toString()
                        + " exists and is not a file");
        } else
            //noinspection ResultOfMethodCallIgnored
            file.createNewFile();
    }

    public static void checkExists(@NonNull File file) throws FileNotFoundException {
        if (!file.exists() || !file.isFile())
            throw new FileNotFoundException("UtilsFileIO -- checkExists: " + file.toString() +
                    " not exists or not is a file");
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/utils/UtilsFileIO.kt
package ru.p3tr0vich.fuel.utils

import java.io.*

object UtilsFileIO {

    @JvmStatic
    @Throws(IOException::class)
    fun read(file: File, strings: MutableList<String>) {
        val fileInputStream = FileInputStream(file)

        val inputStreamReader = InputStreamReader(fileInputStream)

        val bufferedReader = BufferedReader(inputStreamReader)

        var s: String? = bufferedReader.readLine()
        while (s != null) {
            strings.add(s)
            s = bufferedReader.readLine()
        }

        bufferedReader.close()

        inputStreamReader.close()

        fileInputStream.close()
    }

    @JvmStatic
    @Throws(IOException::class)
    fun write(file: File, strings: List<String>) {
        val fileOutputStream = FileOutputStream(file)

        val outputStreamWriter = OutputStreamWriter(fileOutputStream)

        val bufferedWriter = BufferedWriter(outputStreamWriter)

        for (s in strings) {
            bufferedWriter.write(s)
            bufferedWriter.newLine()
        }

        bufferedWriter.close()

        outputStreamWriter.close()

        fileOutputStream.close()
    }

    @JvmStatic
    @Throws(IOException::class)
    fun makeDir(dir: File) {
        if (dir.exists()) {
            return
        }

        if (!dir.mkdirs()) {
            throw IOException("UtilsFileIO -- makeDir: can not create dir " + dir.toString())
        }
    }

    @JvmStatic
    @Throws(IOException::class)
    fun deleteFile(file: File) {
        if (!file.exists()) {
            return
        }

        if (!file.delete()) {
            throw IOException("UtilsFileIO -- deleteFile: can not delete file " + file.toString())
        }
    }

    @JvmStatic
    @Throws(IOException::class)
    fun createFile(file: File) {
        if (file.exists()) {
            if (!file.isFile) {
                throw FileNotFoundException("UtilsFileIO -- createFile: " + file.toString()
                        + " exists and is not a file")
            }
        } else {
            file.createNewFile()
        }
    }

    @JvmStatic
    @Throws(FileNotFoundException::class)
    fun checkExists(file: File) {
        if (!file.exists() || !file.isFile) {
            throw FileNotFoundException("UtilsFileIO -- checkExists: " + file.toString() +
                    " not exists or not is a file")
        }
    }
}